packages/icon/src/vaadin-icon.js

```diff
@@ -226,6 +240,46 @@ class Icon extends ThemableMixin(ElementMixin(ControllerMixin(ConditionalResizeM
     }
   }
 
+  async __srcChanged(src) {
+    if (!src) {
+      return;
+    }
+
+    if (src.includes('#')) {
+      this.svg = svg`<use href="${src}"/>`;
+    } else {
+      try {
+        const svgData = await this.__fetchSVG(src);
+        if (!parser) {
+          parser = new DOMParser();
+        }
+        const parsedResponse = parser.parseFromString(svgData, 'text/html');
+
+        const svgElement = parsedResponse.querySelector('svg');
+        if (!svgElement) {
+          throw new Error(`SVG element not found on path: ${src}`);
+        }
+
+        if (svgElement.hasAttribute('viewBox')) {
+          this.__viewBox = svgElement.getAttribute('viewBox');
+        }
+        this.svg = unsafeSvgLiteral(svgElement.innerHTML);
+      } catch (e) {
+        console.error(e);
+        this.svg = null;
+      }
+    }
+  }
+
+  async __fetchSVG(path) {
+    const data = await fetch(path, { mode: 'cors' });
+    if (!data.ok) {
+      return Promise.reject('Error loading icon');
+    }
```

Comment:
Untested code

Reply:
I think that some of the flakiness in tests are coming from the `fetch` here so I decided to mock this function in tests to `await` on the promise I created inside the test. Otherwise, I could try mocking `fetch` to remove the flakiness.

Reply:
Would it work if you set `this.__fetch = fetch` in the icon constructor and then used `this.__fetch(...)` in the observer function? `icon.__fetch` would be nicely mockable using sinon in the tests.

Reply:
This line is still untested though